docarray/document/mixins/plot.py

```diff
@@ -97,6 +102,110 @@ def display(self):
 
     plot = deprecate_by(display, removed_at='0.5')
 
+    def plot_matching_sprites(
+        self,
+        output: Optional[str] = None,
+        canvas_size: int = 1920,
+        channel_axis: int = -1,
+        top_k: int = 5,
+        min_size: int = 100,
+        image_source: str = 'tensor',
+    ):
+        """Generate a sprite image for the query and its matching images in this Document object.
+
+        An image sprite is a collection of images put into a single image. Query image is on the left
+        followed by matching images. The Document object should contain matches.
+
+        :param output: Optional path to store the visualization. If not given, show in UI
+        :param canvas_size: the width of the canvas
+        :param min_size: the minimum size of the image
+        :param channel_axis: the axis id of the color channel, ``-1`` indicates the color channel info at the last axis
+        :param image_source: specify where the image comes from, can be ``uri`` or ``tensor``. Empty tensor will fallback to uri
+        :param top_k: the number of top matching documents to show in the sprite.
+        """
+        if not self or not self.matches:
+            raise ValueError(f'{self!r} is empty or has no matches')
+
+        if not self.uri and self.tensor is None:
+            raise ValueError(
+                f'Document has neither `uri` nor `tensor`, cannot be plotted'
+            )
+
+        if top_k <= 0:
+            raise ValueError(f'`limit` must be larger than 0, receiving {top_k}')
+
+        if image_source not in ('uri', 'tensor'):
+            raise ValueError(f'image_source can be only `uri` or `tensor`')
+
+        import matplotlib.pyplot as plt
+
+        img_per_row = top_k + 2
+        if top_k > len(self.matches):
+            img_per_row = len(self.matches) + 2
+
+        img_size = int((canvas_size - 50) / img_per_row)
+        if img_size < min_size:
+            # image is too small, recompute the image size and canvas size
+            img_size = min_size
+            canvas_size = img_per_row * img_size + 50
+
+        _d = copy.deepcopy(self)
+        if image_source == 'uri':
+            _d.load_uri_to_image_tensor()
+            channel_axis = -1
```

Comment:
maybe you need `load_uri_to_image_tensor(channel_axis=channel_axis)`? then why do you need `channel_axis` as an argument pass into this function?